src/ray/rpc/rpc_chaos.cc

```diff
@@ -76,35 +76,35 @@ class RpcFailureManager {
       // 25% chance
       num_remaining_failures--;
       return RpcFailure::Request;
-    } else if (rand == 1) {
+    }
+    if (rand == 1) {
       // 25% chance
       num_remaining_failures--;
       return RpcFailure::Response;
-    } else {
-      // 50% chance
-      return RpcFailure::None;
     }
+    // 50% chance
+    return RpcFailure::None;
   }
 
  private:
   absl::Mutex mu_;
   std::mt19937 gen_;
   // call name -> # remaining failures
-  std::unordered_map<std::string, uint64_t> failable_methods_ ABSL_GUARDED_BY(&mu_);
+  absl::flat_hash_map<std::string, uint64_t> failable_methods_ ABSL_GUARDED_BY(&mu_);
 };
 
-static RpcFailureManager _rpc_failure_manager;
+RpcFailureManager rpc_failure_manager;
```

Comment:
Do you want to fix the non-trivially destructible issue in this PR or a followup one? I'm ok with the later :)

Reply:
Let's do it in a follow-up. The absl version we are using doesn't have NoDestructor.

Reply:
btw an easy way to do that is new and never delete

Reply:
```cpp
auto rpc_failure_manager = []() {
  static auto* manager = new RpcFailureManager();
  return *manager;
}();
```

Reply:
For global variables for testing, apart from no destruct, another requirement is lazy initialization (which requires another wrapper); but it's fine here because `RpcFailureManager` is lightweight enough.